xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/pvr/Providers.h

```diff
@@ -0,0 +1,198 @@
+  /// |------|------|----------|----------|-----------
+  /// | **Unique id** | `unsigned int` | @ref PVRProvider::SetUniqueId "SetUniqueId" | @ref PVRProvider::GetUniqueId "GetUniqueId" | *required to set*
+  /// | **Provider name** | `std::string` | @ref PVRProvider::SetName "SetName" | @ref PVRProvider::GetName "GetName" | *required to set*
+  /// | **Provider type** | @ref PVR_PROVIDER_TYPE | @ref PVRProvider::SetType "SetType" | @ref PVRProvider::GetType "GetType" | *optional*
+  /// | **Icon path** | `std::string` | @ref PVRProvider::SetIconPath "SetIconPath" | @ref PVRProvider::GetIconPath "GetIconPath" | *optional*
+  /// | **Countries** | `std::vecotr<std::string>` | @ref PVRProvider::SetCountries "SetCountries" | @ref PVRProvider::GetCountries "GetCountries" | *optional*
+  /// | **Langauges** | `std::vecotr<std::string>` | @ref PVRProvider::SetLangauges "SetLangauges" | @ref PVRProvider::GetLangauges "GetLangauges" | *optional*
+  ///
+
+  /// @addtogroup cpp_kodi_addon_pvr_Defs_PVRProvider
+  ///@{
+
+  /// @brief **required**\n
+  /// Unique identifier for this provider.
+  void SetUniqueId(unsigned int uniqueId) { m_cStructure->iUniqueId = uniqueId; }
+
+  /// @brief To get with @ref SetUniqueId changed values.
+  unsigned int GetUniqueId() const { return m_cStructure->iUniqueId; }
+
+  /// @brief **required**\n
+  /// Name given to this provider.
+  void SetName(const std::string& name)
+  {
+    strncpy(m_cStructure->strName, name.c_str(),
+            sizeof(m_cStructure->strName) - 1);
+  }
+
+  /// @brief To get with @ref SetName changed values.
+  std::string GetName() const { return m_cStructure->strName; }
+
+  /// @brief **optional**\n
+  /// Provider type.
+  ///
+  /// Set to @ref PVR_PVR_PROVIDER_TYPE_UNKNOWN if the type cannot be
+  /// determined.
+  ///
+  /// --------------------------------------------------------------------------
+  ///
+  /// Example:
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// kodi::addon::PVRProvider tag;
+  /// tag.SetType(PVR_PROVIDER_TYPE_SATELLITE);
+  /// ~~~~~~~~~~~~~
+  ///
+  void SetType(PVR_PROVIDER_TYPE type)
+  {
+    m_cStructure->type = type;
+  }
+
+  /// @brief To get with @ref SetType changed values
+  PVR_PROVIDER_TYPE GetType() const { return m_cStructure->type; }
+
+  /// @brief **optional**\n
+  /// Path to the provider icon (if present).
+  void SetIconPath(const std::string& iconPath)
+  {
+    strncpy(m_cStructure->strIconPath, iconPath.c_str(), sizeof(m_cStructure->strIconPath) - 1);
+  }
+
+  /// @brief To get with @ref SetIconPath changed values.
+  std::string GetIconPath() const { return m_cStructure->strIconPath; }
+  ///@}
+
+  /// @brief **optional**\n
+  /// The country codes for the provider.
+  ///
+  /// @note ISO 3166 country codes required (e.g 'GB,IE,CA').
+  void SetCountries(const std::vector<std::string>& countries)
+  {
+    const std::string str = tools::StringUtils::Join(countries, PROVIDER_STRING_TOKEN_SEPARATOR);
```

Comment:
It’s in AddonBase.h. Should it be here too?

Reply:
I would include it here, because it is needed here. It does no harm and we can be sure that we do not need to change this header if for some reason AddonBase.h removes the include.

Reply:
Done